server/module_hash_by_height.go

```diff
@@ -0,0 +1,125 @@
+package server
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/spf13/cobra"
+
+	"cosmossdk.io/store/rootmulti"
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+)
+
+// ModuleHashByHeightQuery retrieves the module hashes at a given height.
+func ModuleHashByHeightQuery[T servertypes.Application](appCreator servertypes.AppCreator[T]) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "module-hash-by-height [height]",
+		Short: "Get module hashes at a given height",
+		Long: `Get module hashes at a given height. This command is useful for debugging and verifying the state of the application at a given height. Daemon should not be running when calling this command.
+Example:
+	appd module-hash-by-height 16841115,
+`,
+		Args: cobra.ExactArgs(1), // Ensure exactly one argument is provided
+		RunE: func(cmd *cobra.Command, args []string) error {
+			heightToRetrieveString := args[0]
+
+			serverCtx := GetServerContextFromCmd(cmd)
+
+			height, err := strconv.ParseInt(heightToRetrieveString, 10, 64)
+			if err != nil {
+				return err
+			}
+
+			commitInfoForHeight, err := getModuleHashesAtHeight(serverCtx, appCreator, height)
+			if err != nil {
+				return err
+			}
+
+			outputFormat, err := cmd.Flags().GetString(flags.FlagOutput)
+			if err != nil {
+				return err
+			}
+
+			switch outputFormat {
+			case flags.OutputFormatJSON:
+				jsonOutput, err := json.MarshalIndent(commitInfoForHeight, "", "  ")
+				if err != nil {
+					return err
+				}
+				cmd.Println(string(jsonOutput))
+			case flags.OutputFormatText:
+				fallthrough
```

Comment:
yaml is used for text type

Reply:
Correct me if I am wrong, but I think after taking your recommendation here https://github.com/cosmos/cosmos-sdk/pull/20779/commits/3dc677580370a48511edca4be011d559736b0d20 we should no longer need this logic branch.